python/ray/experimental/state.py

```diff
@@ -518,109 +518,228 @@ def task_profiles(self, start=None, end=None, num_tasks=None, fwd=True):
+                  full_trace.append(task_trace)
+
+              task = {
+                "cat": "task",
+                "pid": "Node " + str(worker["node_ip_address"]),
+                "tid": info["worker_id"],
+                "id": str(task_id),
+                "ts": micros_rel(info["get_arguments_start"]),
+                "ph": "X",
+                "name": info["function_name"],
+                "args": total_info,
+                "dur": micros(info["store_outputs_end"] -
+                              info["get_arguments_start"]),
+                "cname": "thread_state_runnable"
+              }
+              full_trace.append(task)
+
+          if task_dep:
+              if parent_info:
+                  parent_worker = workers[parent_info["worker_id"]]
+                  parent_times = self._get_times(parent_info)
+                  parent = {
+                      "cat": "submit_task",
+                      "pid": "Node " + str(parent_worker["node_ip_address"]),
+                      "tid": parent_info["worker_id"],
+                      "ts": micros_rel(task_profiles[task_table[task_id]["TaskSpec"]["ParentTaskID"]]["get_arguments_start"]),
+                      "ph": "s",
+                      "name": "SubmitTask",
+                      "args": {},
+                      "id": str(worker) + str(micros(min(parent_times)))
+                  }
+                  full_trace.append(parent)
+                  task_trace = {
+                     "cat": "submit_task",
+                     "pid": "Node " + str(worker["node_ip_address"]),
+                     "tid": info["worker_id"],
+                     "ts": micros_rel(info["get_arguments_start"]),
+                     "ph": "f",
+                     "name": "SubmitTask",
+                     "args": {},
+                     "id": str(worker) + str(micros(min(parent_times))),
+                     "bp": "e",
                     }
-                    full_trace.append(execute_trace)
-            else:
-                task = {
-                  "cat": "task",
-                  "pid": "Node " + str(worker["node_ip_address"]),
-                  "tid": info["worker_id"],
-                  "id": str(worker),
-                  "ts": micros_rel(info["get_arguments_start"]),
-                  "ph": "X",
-                  "name": info["function_name"],
-                  "args": delta_info,
-                  "dur": micros(info["store_outputs_end"] -
-                                info["get_arguments_start"])
-                }
-                full_trace.append(task)
-
-        print("dumping {}/{}".format(len(full_trace), len(task_info)))
-        with open(path, "w") as outfile:
-            json.dump(full_trace, outfile)
+                  full_trace.append(task_trace)
+
+          if obj_dep: 
+              args = task_table[task_id]["TaskSpec"]["Args"]
+              for arg in args:
+                  if type(arg) is int:
```

Comment:
why do we have a special case for integers? shouldn't we be checking if `isinstance(arg, ray.local_scheduler.ObjectID)`?